cmd/thanos/compact.go

```diff
@@ -40,6 +39,8 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	syncDelay := cmd.Flag("sync-delay", "Minimum age of fresh (non-compacted) blocks before they are being processed.").
 		Default("30m").Duration()
 
+	retention := modelDuration(cmd.Flag("storage.retention", "How long to retain samples in storage. 0d - disables retention").Default("0d"))
```

Comment:
hm I would actually think about future here.. if people really want per -resolution retention is this flag sufficient for it or we will break compatibility by extending this in future?

Reply:
Can we mention that this is not more granurar than potentially lenght of the most compacted block? (2w currently)